ref-exchange/src/lib.rs

```diff
@@ -193,6 +188,8 @@ impl Contract {
         deposits.add(token_out.as_ref().clone(), amount_out);
         self.deposited_amounts.insert(&sender_id, &deposits);
         self.pools.replace(pool_id, &pool);
+        deposits.update_storage(start_storage);
+        self.deposited_amounts.insert(&sender_id, &deposits);
```

Comment:
To verify: I think we need to insert 2 times:
* once in line 194 (prev)
* second time here.
This is because we won't trace properly new storage until we insert the record into the storage tree.
Alternative would be to refactor the AccountDeposit and move `ynear` to another, top-level map.